tests/reference/cpp-doconcurrentloop_01-4e9f274.stdout

```diff
@@ -28,8 +28,8 @@ int main()
     scalar = 10;
     nsize = 1234;
     Kokkos::parallel_for(Kokkos::RangePolicy<Kokkos::DefaultExecutionSpace>(1, nsize - 1+1), KOKKOS_LAMBDA(const long i) {
-        a[i-1] = 5;
-        b[i-1] = 5;
+        a[i + 1 - 1] = 5;
+        b[i + 1 - 1] = 5;
```

Comment:
I think this change is not correct.

Reply:
The `i` runs from 1 to `nsize` (I think) and the array `a` is indexed from 0 to 10000-1.

Reply:
Yeah, that's true for this case.
But this fails for the following case:
Py code:
```
def test_Set():
    a: i32
    b: i32

    a = {1, 2, 3}
    a = {2, 3, 4, 5, 5}
    a = {"a", "b", "c"}
    b = a[0]
```

CPP:
```cpp
void test_Set()
{
    int a;
    int b;
    a = {1, 2, 3};
    a = {2, 3, 4, 5, 5};
    a = {"a", "b", "c"};
    b = a[0-1];
}
```

Reply:
That's because the "a" should not be an array ref, it's not an array.

Reply:
Yep, make sense!